tests/unit/test_waiters.py

```diff
@@ -342,6 +342,34 @@ def test_unspecified_errors_stops_waiter(self):
         with self.assertRaises(WaiterError):
             waiter.wait()
 
+    def test_unspecified_errors_propagate_error_code(self):
+        # If a waiter receives an error response, then the
+        # waiter should pass along the error code
+        config = self.create_waiter_config()
+        operation_method = mock.Mock()
+        error_code = 'error_message'
+        error_message = 'error_message'
+        self.client_responses_are(
+            # This is an unknown error that's not called out
+            # in any of the waiter config, so when the
+            # waiter encounters this response it will transition
+            # to the failure state.
+            {'Error': {'Code': error_code, 'Message': error_message}},
+            for_operation=operation_method
+        )
+        waiter = Waiter('MyWaiter', config, operation_method)
+
+        exception = None
+        try:
+            waiter.wait()
+        except WaiterError as e:
```

Comment:
Oh for this type of logic, I like using [assertRaisesRegexp](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp). It makes the assertion much shorter and nicer.